src/subclass/container.rs

```diff
@@ -206,3 +221,17 @@ where
 
     imp.get_path_for_child(&wrap, &widget).to_glib_none().0
 }
+
+unsafe extern "C" fn container_forall<T: ObjectSubclass, P: FnMut(&Widget)>(
+    ptr: *mut gtk_sys::GtkContainer,
+    callback: P
```

Comment:
... because then you would see that the signature here is wrong :) It should be `fn(*mut GtkContainer, gboolean, GtkCallback, gpointer)`

Reply:
Hm - can you clarify this a bit more? I really struggle to understand how this works. I had the hope it's a bit easier to implement :/

```
unsafe extern "C" fn container_forall<T: ObjectSubclass, P: FnMut(&Widget)>(
    ptr: *mut gtk_sys::GtkContainer,
    include_internals: gboolean,
    callback: P,
    user_data: gpointer,
)
```

- What should be the last `gpointer`? `user_data`?
- For `callback` I have currently `P: FnMut(&Widget)`. That should be get replaced with `GtkCallback`?

Reply:
I also have a hard time remembering how this worked after that long time, will have to read it up again :) It's not a simple case unfortunately.